src/libasr/codegen/x86_assembler.h

```diff
@@ -567,6 +641,21 @@ class X86Assembler {
         EMIT("mov " + r2s(r32) + ", " + i2s(imm32));
     }
 
+    void rex(int W, int R, int X, int B) {
+        LFORTRAN_ASSERT(W <= 1);
+        LFORTRAN_ASSERT(R <= 1);
+        LFORTRAN_ASSERT(X <= 1);
+        LFORTRAN_ASSERT(B <= 1);
+        m_code.push_back(m_al, (0b01000000 | (W << 3) | (R << 2) | (X << 1) | B));
+    }
+
+    void asm_mov_r64_imm64(X64Reg r64, uint64_t imm64) {
+        rex(1, 0, 0, 0);
```

Comment:
I would write this as:
```suggestion
        m_code.push_back(m_al, rex(1, 0, 0, 0));
```

Reply:
Got it. It looks great. Yes, it makes it explicit and much simpler. Thank you so much for sharing.